azext_iot/tests/test_iot_ext_int.py

```diff
@@ -719,9 +719,7 @@ def test_hub_device_twins(self):
 
         # Region specific test
         if self.region not in ["West US 2", "North Europe", "Southeast Asia"]:
-            warnings.warn(
-                "Skipping distributed-tracing tests. IoT Hub not in supported region!"
-            )
+            warnings.warn(UserWarning("Skipping distributed-tracing tests. IoT Hub not in supported region!"))
```

Comment:
should we be using UserWarning elsewhere too? What benefits does it give us?

Reply:
The benefit here is pytest will automatically captures warnings from tests and shows them after the run. This was put in place to explain why some inner tests were skipped and the reason why. This pattern was used rather than say a `pytest.skip` or similar because of the way the tests were written and not wanting to show the whole test skipped when only the last portion is skipped.

Reply:
At the user interaction layer, if you want to show a warning logger.warning() is what you want to use.